### ISSUE TITLE
Allow configuring a maximum number of nodes put into orchestrated maintenance mode

### ISSUE THREAD

**Issue by wardsng** [vespa-engine/vespa#11911]
When using the State API to put a storage node into safe maintenance mode, only one node is allowed in such a state, regardless of the redundancy levels we have set.

For example, in a 6x redundancy 3-group setup (2x redundancy per group) we would expect to be able to set a node in group0 and a node in group1 into maintenance mode.

**Comment by vekterli** [comment]
While on the subject, another nice to have™ feature for orchestrated node state transitions would be the ability to set entire _groups_ to maintenance. This would allow for upgrading _all_ nodes in the group in parallel at the cost of transient reduced group availability.

**Comment by johans1** [comment]
We assume this is no longer relevant. Please reopen if still valid.
